### DISCUSSION TITLE
Issues with Teampass Installation on Ubuntu 24 (PHP 7.4, PHP 8.1) - Multiple Versions Tested (3.1.1, 3.1.2)

### DISCUSSION THREAD

**Discussion by martysouza** [nilsteampassnet/TeamPass#4521] [General]
Subject: Issues with Teampass Installation on Ubuntu 24 (PHP 7.4, PHP 8.1) - Multiple Versions Tested (3.1.1, 3.1.2)

Message:

Hello,

We have been trying to install Teampass on a fresh Ubuntu 24 system but have encountered several issues during the installation process. Here is a detailed summary of our environment, steps taken, and the problems we faced:

Environment Details:
Operating System: Ubuntu 24
PHP Versions Tested: 8.1.2, 7.4
Teampass Versions Tested: 3.1.1, 3.1.2
Web Server: Apache 2
Database: MariaDB 10.6
Steps Taken and Issues Encountered:
1. Installation with Teampass 3.1.1 and PHP 8.1.2
The installation failed with the error: "Answer from server is empty".
Debugging revealed a 500 Internal Server Error when accessing the install.queries.php file.
Upon investigating, we found missing PHP extensions (ext-bcmath and ext-ldap), which were installed.
Despite fixing the dependencies, the installation still failed.
2. Downgrade to PHP 7.4 for Compatibility
Given the potential compatibility issues with PHP 8.1, we removed PHP 8.1.2 and installed PHP 7.4.
Reinstalled all required PHP extensions:

sudo apt-get install php7.4-mbstring php7.4-curl php7.4-json php7.4-xml php7.4-mysqli php7.4-zip php7.4-gd php7.4-bcmath php7.4-ldap
Verified that Teampass dependencies were installed using Composer:

composer install
Even with PHP 7.4, the 500 Internal Server Error persisted.
3. Testing Teampass 3.1.2
Upgraded to Teampass 3.1.2 to see if the newer version resolved the issues.
Faced the same "Answer from server is empty" error during installation.
The install.queries.php file could not be executed successfully, resulting in an HTTP 500 error.
4. Other Troubleshooting Steps
Verified database connection:
sql
Copiar código
CREATE DATABASE teampass CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;
CREATE USER 'teampass_user'@'localhost' IDENTIFIED BY 'secure_password';
GRANT ALL PRIVILEGES ON teampass.* TO 'teampass_user'@'localhost';
Successfully connected to the database manually but the installation process did not proceed.
Ensured proper file permissions:
bash
Copiar código
sudo chown -R www-data:www-data /var/www/html/teampass
sudo chmod -R 755 /var/www/html/teampass
Enabled PHP error reporting:
ini
Copiar código
display_errors = On
error_reporting = E_ALL
Checked logs (/var/log/apache2/error.log) but found only generic HTTP 500 errors.
Current Status
Neither Teampass 3.1.1 nor 3.1.2 could be installed successfully.
The main issues revolve around the "Answer from server is empty" error and HTTP 500 errors during the install.queries.php execution.
We suspect there might be compatibility issues with Ubuntu 24 or missing steps in the installation process for newer PHP versions.
Questions:
Is there a recommended PHP version for Teampass 3.1.1 or 3.1.2 on Ubuntu 24?
Are there known issues or patches for compatibility with newer distributions and PHP versions?
Could there be additional dependencies or configurations we might have missed?
We appreciate any guidance or suggestions to resolve these issues. Let us know if additional logs or information are required.

Thank you!
![erro_tempass](https://github.com/user-attachments/assets/779b9f81-3f3b-44b6-8dc9-e66c4fa1c97f)

**Comment by martysouza** [comment]
??

**Comment by DobroFenix** [comment]
Use latest version.
You launch not working version.
